legal-api/src/legal_api/models/db.py

```diff
@@ -11,17 +11,206 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-"""Create SQLAlchenmy and Schema managers.
+"""Create SQLAlchemy and Schema managers.
 
 These will get initialized by the application using the models
 """
-from flask_sqlalchemy import SQLAlchemy
-from sqlalchemy_continuum import make_versioned
+import functools
+from functools import wraps
+
+from flask import current_app
+from flask_sqlalchemy import SignallingSession, SQLAlchemy
+from sqlalchemy import event, orm
+from sqlalchemy.orm import Session, mapper
+from sqlalchemy_continuum import make_versioned, remove_versioning
+from sqlalchemy_continuum import version_class as _old_version_class
+from sqlalchemy_continuum import versioning_manager
+
+# from sql_versioning.versioned_history import versioned_session
+from sql_versioning.versioned_history import TransactionManager
+from sql_versioning.versioned_history import disable_versioning as _disable_new_versioning
+from sql_versioning.versioned_history import enable_versioning as _enable_new_versioning
+from sql_versioning.versioned_history import versioned_cls as _new_version_class
 
 
 # by convention in the Flask community these are lower case,
 # whereas pylint wants them upper case
 db = SQLAlchemy()  # pylint: disable=invalid-name
 
 # make_versioned(user_cls=None, plugins=[FlaskPlugin()])
-make_versioned(user_cls=None)
+# make_versioned(user_cls=None)
+
+
+def init_db(app):
+    db.init_app(app)
+    orm.configure_mappers()
+
+
+def debug(func):
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        print(f'\033[34m--> Entering {func.__qualname__}()\033[0m')
+        ret = func(*args, **kwargs)
+        print(f'\033[34m<-- Exiting {func.__qualname__}()\033[0m')
+        return ret
+
+    return wrapper
+
+
+def _enable_old_versioning():
+    versioning_manager.track_operations(mapper)
+    versioning_manager.track_session(Session)
+
+def _disable_old_versioning():
+    versioning_manager.remove_operations_tracking(mapper)
+    versioning_manager.remove_session_tracking(Session)
+
+def _old_get_transaction_id(session):
+    uow = versioning_manager.unit_of_work(session)
+    transaction = uow.create_transaction(session)
+    return transaction.id
+
+def _new_get_transaction_id(session):
+    new_transaction_manager = TransactionManager(session)
+    return new_transaction_manager.create_transaction()
+
+
+class VersioningProxy:
```

Comment:
I change to use proxy instead of custom db session to dispatch two versionings. Hope it will be clearer.